rewriter/Minitest.cc

```diff
@@ -145,7 +178,42 @@ string to_s(core::Context ctx, unique_ptr<ast::Expression> &arg) {
     return arg->toString(ctx);
 }
 
-unique_ptr<ast::Expression> runSingle(core::MutableContext ctx, ast::Send *send) {
+// The 'context' here is any expressions which are put at the top of the 'it' blocks. If we've looked at an instance of
+// `test_each`, then we need to somehow pull the variable being introduced by the `each` into the body of the method. We
+// do that by taking
+//
+//   test_each(expr) do |var|
+//
+// and turning it into an assignment like
+//
+//   var = T.must(expr.collect.first)
+//
+// which will have both the correct type and the correct name. (The reason for the `collect` is that if `expr` is an
+// array literal, then Sorbet would treat it as a tuple type, and `first` would be the first thing in the tuple,
+// i.e. `[1,true].first` has the type `Integer`, but we're using this to stand in for an arbitrary member of the
+// collection, not actually the first one: what we want is something with the type `T.any(Integer,TrueClass)` here. The
+// `.collect` will give us an array instead of a tuple, then `first` will give us a nilable element, and then the `must`
+// will remove the nilable.
+unique_ptr<ast::Expression> makeContext(core::MutableContext ctx, unique_ptr<ast::Expression> blockArg,
+                                        unique_ptr<ast::Expression> &as, optional<ast::Expression *> context) {
```

Comment:
Let's rename `context`. I'd like to reserve that name for variables that are of type `core::Context` or `core::MutableContext`. Something like `fooContext` would be ok though.